src/commands/nsec3hash.rs

```diff
@@ -1,55 +1,61 @@
-use crate::error::Error;
+use std::io::Write;
+use std::str::FromStr;
+
 use clap::builder::ValueParser;
 use domain::base::iana::nsec3::Nsec3HashAlg;
 use domain::base::name::{self, Name};
-use domain::base::ToName;
-use domain::rdata::nsec3::{Nsec3Salt, OwnerHash};
+use domain::rdata::nsec3::Nsec3Salt;
+use domain::validate::nsec3_hash;
 use lexopt::Arg;
-// use domain::validator::nsec::nsec3_hash;
-use octseq::OctetsBuilder;
-use ring::digest;
-use std::str::FromStr;
+
+use crate::error::Error;
 
 use super::{parse_os, parse_os_with, LdnsCommand};
 
 #[derive(Clone, Debug, clap::Args)]
+#[cfg_attr(feature = "arbitrary", derive(arbitrary::Arbitrary))]
 pub struct Nsec3Hash {
     /// The hashing algorithm to use
     #[arg(
-        long,
+        long = "algorithm",
         short = 'a',
-        value_name = "NUMBER_OR_MNEMONIC",
-        default_value_t = Nsec3HashAlg::SHA1,
-        value_parser = ValueParser::new(Nsec3Hash::parse_nsec_alg)
+        value_name = "NUMBER OR MNEMONIC",
+        default_value = "SHA-1",
+        value_parser = ValueParser::new(Nsec3Hash::parse_nsec3_alg)
     )]
     algorithm: Nsec3HashAlg,
 
     /// The number of hash iterations
     #[arg(
-        long,
+        long = "iterations",
         short = 'i',
         visible_short_alias = 't',
         value_name = "NUMBER",
-        default_value_t = 1
+        default_value_t = 0
     )]
     iterations: u16,
 
     /// The salt in hex representation
-    #[arg(short = 's', long, value_name = "HEX_STRING", default_value_t = Nsec3Salt::empty())]
+    #[arg(
+        long = "salt",
+        short = 's',
+        value_name = "HEX STRING",
+        default_value_t = Nsec3Salt::empty(),
+        value_parser = ValueParser::new(Nsec3Hash::parse_salt)
+    )]
     salt: Nsec3Salt<Vec<u8>>,
 
     /// The domain name to hash
-    #[arg(value_name = "DOMAIN_NAME", value_parser = ValueParser::new(Nsec3Hash::parse_name))]
+    #[arg(value_name = "DOMAIN NAME", value_parser = ValueParser::new(Nsec3Hash::parse_name))]
     name: Name<Vec<u8>>,
 }
 
 const LDNS_HELP: &str = "\
 ldns-nsec3-hash [OPTIONS] <domain name>
   prints the NSEC3 hash of the given domain name
-
-  -a <algorithm> hashing algorithm number
-  -t <number>    iterations
-  -s <string>    salt in hex\
+-a [algorithm] hashing algorithm
+-t [number] number of hash iterations
+-s [string] salt
```

Comment:
The indentation was intentional because I liked it better, but I guess for compat this would be more "correct". The angle brackets were "more correct" (for some definition of correct based on convention) because they indicate a required value where square brackets mean optional values.

Reply:
I'm reluctant to start improving the ldns-xxx commands at this stage, I'd rather get things implemented and basically working and as compatible as we can easily do. The temptation to start tweaking things if we start might be great and where do we stop? :-)